File: backend/src/main/java/com/dashboard/entity/Notification.java
package com.dashboard.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "notifications", indexes = {
        @Index(name = "idx_notification_recipient", columnList = "recipient_id"),
        @Index(name = "idx_notification_read", columnList = "is_read"),
        @Index(name = "idx_notification_type", columnList = "type"),
        @Index(name = "idx_notification_created", columnList = "created_at")
})
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Notification {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "recipient_id", nullable = false)
    private User recipient;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 30)
    private NotificationType type;

    @Column(nullable = false, length = 255)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String message;

    @Column(name = "reference_id")
    private String referenceId;

    @Column(name = "reference_type", length = 50)
    private String referenceType;

    @Column(name = "action_url", length = 500)
    private String actionUrl;

    @Column(name = "is_read")
    @Builder.Default
    private Boolean isRead = false;

    @Column(name = "read_at")
    private LocalDateTime readAt;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    public enum NotificationType {
        PRODUCT_APPROVED("Produit approuvé"),
        PRODUCT_REJECTED("Produit rejeté"),
        PRODUCT_PURCHASED("Produit acheté"),
        REVIEW_RECEIVED("Avis reçu"),
        RATING_RECEIVED("Note reçue"),
        ORDER_CONFIRMED("Commande confirmée"),
        ORDER_SHIPPED("Commande expédiée"),
        ORDER_DELIVERED("Commande livrée"),
        NEW_ORDER("Nouvelle commande"),
        NEW_SELLER_PRODUCT("Nouveau produit vendeur"),
        NEW_SELLER_REGISTRATION("Nouvelle inscription vendeur"),
        STOCK_ALERT("Stock Alert"),
        SYSTEM("Système"),
        PROMOTION("Promotion"),
        PREDICTION_BESTSELLER("Bestseller potentiel détecté"),
        PREDICTION_PRICE("Recommandation de prix");

        private final String description;

        NotificationType(String description) {
            this.description = description;
        }

        public String getDescription() {
            return description;
        }
    }

    public void markAsRead() {
        this.isRead = true;
        this.readAt = LocalDateTime.now();
    }
}

File: frontend/Main/src/service/notification.ts
/**
 * Notification Types - MouadVision Platform
 * Includes ML Prediction Notification Types
 */

export type NotificationType =
    | 'PRODUCT_APPROVED'
    | 'PRODUCT_REJECTED'
    | 'PRODUCT_PURCHASED'
    | 'REVIEW_RECEIVED'
    | 'RATING_RECEIVED'
    | 'ORDER_CONFIRMED'
    | 'ORDER_SHIPPED'
    | 'ORDER_DELIVERED'
    | 'NEW_ORDER'
    | 'NEW_SELLER_PRODUCT'
    | 'NEW_SELLER_REGISTRATION'
    | 'STOCK_ALERT'
    | 'SYSTEM'
    | 'PROMOTION'
    | 'PREDICTION_BESTSELLER'  // 🔮 ML Prediction
    | 'PREDICTION_PRICE';       // 💹 ML Price Recommendation

export interface Notification {
    id:number;
    type:NotificationType | string;
    typeDescription:string;
    title:string;
    message:string;
    referenceId:string;
    referenceType:string;
    actionUrl:string;
    isRead:boolean;
    readAt:string | null;
    createdAt:string;
}

export interface NotificationState {
    notifications:Notification[];
    unreadCount:number;
    loading:boolean;
    error:string | null;
}

export interface NotificationIconConfig {
    icon:string;
    bg:string;
    gradient?:string;
    textColor?:string;
}

export const NOTIFICATION_ICONS:Record<string, NotificationIconConfig> = {
    PRODUCT_APPROVED:{ icon:'✅', bg:'bg-green-100 dark:bg-green-900/30', textColor:'text-green-600' },
    PRODUCT_REJECTED:{ icon:'❌', bg:'bg-red-100 dark:bg-red-900/30', textColor:'text-red-600' },
    PRODUCT_PURCHASED:{ icon:'💰', bg:'bg-yellow-100 dark:bg-yellow-900/30', textColor:'text-yellow-600' },
    REVIEW_RECEIVED:{ icon:'⭐', bg:'bg-purple-100 dark:bg-purple-900/30', textColor:'text-purple-600' },
    RATING_RECEIVED:{ icon:'🌟', bg:'bg-amber-100 dark:bg-amber-900/30', textColor:'text-amber-600' },
    ORDER_CONFIRMED:{ icon:'📦', bg:'bg-blue-100 dark:bg-blue-900/30', textColor:'text-blue-600' },
    ORDER_SHIPPED:{ icon:'🚚', bg:'bg-cyan-100 dark:bg-cyan-900/30', textColor:'text-cyan-600' },
    ORDER_DELIVERED:{ icon:'✔️', bg:'bg-emerald-100 dark:bg-emerald-900/30', textColor:'text-emerald-600' },
    NEW_ORDER:{ icon:'🛒', bg:'bg-indigo-100 dark:bg-indigo-900/30', textColor:'text-indigo-600' },
    NEW_SELLER_PRODUCT:{ icon:'🆕', bg:'bg-indigo-100 dark:bg-indigo-900/30', textColor:'text-indigo-600' },
    NEW_SELLER_REGISTRATION:{ icon:'👤', bg:'bg-violet-100 dark:bg-violet-900/30', textColor:'text-violet-600' },
    STOCK_ALERT:{ icon:'📊', bg:'bg-orange-100 dark:bg-orange-900/30', textColor:'text-orange-600' },
    SYSTEM:{ icon:'🔔', bg:'bg-gray-100 dark:bg-gray-700', textColor:'text-gray-600' },
    PROMOTION:{ icon:'🎉', bg:'bg-pink-100 dark:bg-pink-900/30', textColor:'text-pink-600' },
    // 🔮 ML Prediction Notifications
    PREDICTION_BESTSELLER:{
        icon:'🔮',
        bg:'bg-gradient-to-br from-yellow-100 to-orange-100 dark:from-yellow-900/30 dark:to-orange-900/30',
        gradient:'from-yellow-500 to-orange-500',
        textColor:'text-orange-600'
    },
    PREDICTION_PRICE:{
        icon:'💹',
        bg:'bg-gradient-to-br from-blue-100 to-purple-100 dark:from-blue-900/30 dark:to-purple-900/30',
        gradient:'from-blue-500 to-purple-500',
        textColor:'text-purple-600'
    },
};

export const getNotificationIcon = (type:string):NotificationIconConfig => {
    return NOTIFICATION_ICONS[type] || NOTIFICATION_ICONS.SYSTEM;
};

export const isPredictionNotification = (type:string):boolean => {
    return type === 'PREDICTION_BESTSELLER' || type === 'PREDICTION_PRICE';
};